sbi/mcmc/init_strategy.py

```diff
@@ -23,25 +26,32 @@ def sir(
     Args:
         prior: Prior distribution, candidate samples are drawn from it.
         potential_fn: Potential function that the candidate samples are weighted with.
+            Note that the function needs to return log probabilities.
         init_strategy_num_candidates: Number of candidate samples drawn.
+        batch_size: Batch size used for evaluating candidates.
 
     Returns:
         A single sample.
     """
-
-    init_param_candidates = prior.sample((init_strategy_num_candidates,)).detach()
-
-    log_weights = torch.cat(
-        [
-            potential_fn(init_param_candidates[i, :]).detach()
-            for i in range(init_strategy_num_candidates)
-        ]
-    )
-    probs = np.exp(log_weights.view(-1).numpy().astype(np.float64))
-    probs[np.isnan(probs)] = 0.0
-    probs[np.isinf(probs)] = 0.0
-    probs /= probs.sum()
-    idxs = np.random.choice(
-        a=np.arange(init_strategy_num_candidates), size=1, replace=False, p=probs,
-    )
-    return init_param_candidates[torch.from_numpy(idxs.astype(int)), :]
+    with torch.set_grad_enabled(False):
+        num_batches = int(init_strategy_num_candidates / batch_size)
```

Comment:
Just realized that this will break if `init_strategy_num_candidates` is not a multiple of `batch_size`. Importantly, if `init_strategy_num_candidates` is smaller than `batch_size` (default 1000), casting to `int` will truncate and `num_batches` will be 0. This should be fixed.